internal/pkg/cli/deploy/rdws.go

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package deploy
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/copilot-cli/internal/pkg/apprunner"
+	awsapprunner "github.com/aws/copilot-cli/internal/pkg/aws/apprunner"
+	"github.com/aws/copilot-cli/internal/pkg/aws/partitions"
+	"github.com/aws/copilot-cli/internal/pkg/aws/s3"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/upload/customresource"
+	"github.com/aws/copilot-cli/internal/pkg/describe"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/aws/copilot-cli/internal/pkg/template"
+	"github.com/aws/copilot-cli/internal/pkg/term/color"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+)
+
+var rdwsAliasUsedWithoutDomainFriendlyText = fmt.Sprintf("To use %s, your application must be associated with a domain: %s.\n",
+	color.HighlightCode("http.alias"),
+	color.HighlightCode("copilot app init --domain example.com"))
+
+type rdwsDeployer struct {
+	*svcDeployer
+	customResourceS3Client uploader
+	appVersionGetter       versionGetter
+	rdwsMft                *manifest.RequestDrivenWebService
+	customResources        customResourcesFunc
+}
+
+// NewRDWSDeployer is the constructor for RDWSDeployer.
+func NewRDWSDeployer(in *WorkloadDeployerInput) (*rdwsDeployer, error) {
+	svcDeployer, err := newSvcDeployer(in)
+	if err != nil {
+		return nil, err
+	}
+	versionGetter, err := describe.NewAppDescriber(in.App.Name)
+	if err != nil {
+		return nil, fmt.Errorf("new app describer for application %s: %w", in.App.Name, err)
+	}
+	rdwsMft, ok := in.Mft.(*manifest.RequestDrivenWebService)
+	if !ok {
+		return nil, fmt.Errorf("manifest is not of type %s", manifest.RequestDrivenWebServiceType)
+	}
+	return &rdwsDeployer{
+		svcDeployer:            svcDeployer,
+		customResourceS3Client: s3.New(svcDeployer.defaultSessWithEnvRegion),
+		appVersionGetter:       versionGetter,
+		rdwsMft:                rdwsMft,
+		customResources: func(fs template.Reader) ([]*customresource.CustomResource, error) {
+			crs, err := customresource.RDWS(fs)
+			if err != nil {
+				return nil, fmt.Errorf("read custom resources for a %q: %w", manifest.RequestDrivenWebServiceType, err)
+			}
+			return crs, nil
+		},
+	}, nil
+}
+
+// IsServiceAvailableInRegion checks if service type exist in the given region.
+func (rdwsDeployer) IsServiceAvailableInRegion(region string) (bool, error) {
+	return partitions.IsAvailableInRegion(awsapprunner.EndpointsID, region)
+}
+
+// UploadArtifacts uploads the deployment artifacts such as the container image, custom resources, addons and env files.
+func (d *rdwsDeployer) UploadArtifacts() (*UploadArtifactsOutput, error) {
+	return d.uploadArtifacts(d.customResources)
+}
+
+type rdwsDeployOutput struct {
```

Comment:
Can we move this struct definition and its `RecommendedAction` to just before `DeployWorkload` or to after all of `rdwsDeployer`'s public methods?